recipes/xkbcommon/all/conanfile.py

```diff
@@ -70,80 +65,97 @@ def validate(self):
             raise ConanInvalidConfiguration("This library is only compatible with Linux or FreeBSD")
 
     def build_requirements(self):
-        self.build_requires("meson/0.63.1")
-        self.build_requires("bison/3.7.6")
-        if hasattr(self, "settings_build") and self.options.get_safe("wayland"):
-            self.build_requires("wayland/1.21.0")
+        self.tool_requires("meson/0.63.1")
+        self.tool_requires("bison/3.7.6")
+        self.tool_requires("pkgconf/1.7.4")
+        if self.options.get_safe("with_wayland"):
```

Comment:
Not a good idead to remove `hasattr(self, "settings_build")` check. It prevents dependencies conflict with 1 profile when wayland version is overridden.

Reply:
Added back in #12882